_drafts/wsl2-firmware-development.md

```diff
@@ -0,0 +1,655 @@
+    up to your Microsoft account
+  - unsigned applications and scripts produce a warning (this is also a con)
+- Hardware Support: if your product doesn't support Windows on release then it's
+  dead on arrival.
+- Software Support: mechanical and electrical engineering tools have long been
+  built for Windows.
+- Your clients probably use it.  Developing and testing software on the target
+  platform is generally a good idea.
+- Stability.  Freezes and crashes are all but nonexistent.  Of course, that’s
+  largely anecdotal, as are the instabilities associated with desktop Linux.
+
+Cons:
+- Zephyr performance: half the speed of Linux.
+- Security:
+  - Developer mode removes warnings about unsigned scripts and applications.
+  - Microsoft requires yearly payment — previously, to an approved certificate
+  authority, now you have the option of paying MS directly — in order to sign scripts
+  and applications.  This has resulted in the widespread use of unsigned scripts
+  and applications.  Even huge projects like Rye and WezTerm will trigger false
+  positives for Trojans in Windows Defender and require the user to click through
+  “scary warnings”.  The global software ecosystem lives and dies by OSS and
+  creating barriers to signing and authentication of OSS is dangerous for
+  Microsoft’s customers and business.
+- It’s $200, but you might never notice that when purchasing new laptops.
+- It’s Windows.  This is more of a “gut feeling” thing that I can best explain 
+  with an anecdote.
+  
+  As I was setting up my new laptop, I used my Microsoft
+  account when installing Windows, something that I had long avoided, not really
+  for any specific reason.  It felt like I could potentially avoid “Windows problems”
+  by being compliant here.  It went OK, except that it never prompted me for the
+  name of my home (AKA user) folder and then it named it incorrectly: “jphut” instead of
+  “jp”.  Now, many people would have left it there.  And perhaps I should have. 
+  But “jphut” is _wrong_.  Home is at `/home/jp`, or `c:/users/jp`, or `/users/jp` -
+  this is how it has _always been_.  “No big deal”, I thought. “I’ll fix it!” 
+  
+  It took me 3 hours.  RegEdit.  Reviewing scripts left on Windows forums. 
+  Seriously breaking my install.  Adding symlinks for OneDrive.  It’s been working
+  fine for a few months now, but for all I know I’m out of spec in a weird way that
+  will cause problems.  So, to quote Linus Torvalds, “F@&# you, \[Microsoft\]!”
+- 📎Unsettling paperclip mascot!
+
+#### VMWare Workstation Pro
+
+⭐⭐⭐
+
+
+Pros:
+- With a fully featured VM, you get most of the pros of Linux with access to Window’s strong points when you need them.
+- USB forwarding, including BLE, works well because you can run a full Linux distro of your choice with all the kernel modules that you need.
+- SSH to a headless VM allows for a snappy, low-latency, experience in your
+  editor of choice.
+
+Cons:
```

Comment:
Isn't cost a con? And long term support is iffy

Reply:
What do we know about long term support?  I think I had already switch to WSL2 when I heard people talking about something.

Agreed re: cost, and I'll address that it's "free for personal use" now: https://blogs.vmware.com/workstation/2024/05/vmware-workstation-pro-now-available-free-for-personal-use.html

Reply:
It took me over an hour to actually find the download link for VMWare recently. Eventually Dave just emailed me one he kept. It made me think that since VMWare got bought by Broadcom they purposly made it difficult to download the free version and would eventually deprecate it. Just my thought, nothing concrete.